pkg/collector/collector.go

```diff
@@ -0,0 +1,72 @@
+package collector
+
+import (
+	"bytes"
+	"io"
+	"mime/multipart"
+	"net/http"
+	"os"
+)
+
+func createSendToCollectorPostRequest(endPoint, claimFilePath, createdBy, partnerName string) (*http.Request, error) {
+	// Create a new buffer to hold the form-data
+	var buffer bytes.Buffer
+	w := multipart.NewWriter(&buffer)
+
+	// Add the claim file to the request
+	claimFile, err := os.Open(claimFilePath)
+	if err != nil {
+		return nil, err
+	}
+	defer claimFile.Close()
+	fw, err := w.CreateFormFile("claimFile", claimFilePath)
+	if err != nil {
+		return nil, err
+	}
+	if _, err = io.Copy(fw, claimFile); err != nil {
+		return nil, err
+	}
+
+	// Add the created by and partner name to the request
+	if fw, err = w.CreateFormField("created_by"); err != nil {
+		return nil, err
+	}
+	if _, err = fw.Write([]byte(createdBy)); err != nil {
+		return nil, err
+	}
+
+	if fw, err = w.CreateFormField("partner_name"); err != nil {
+		return nil, err
+	}
+	if _, err = fw.Write([]byte(partnerName)); err != nil {
+		return nil, err
+	}
+
+	w.Close()
+
+	// Create POST request with the form-data as body
+	req, err := http.NewRequest("POST", endPoint, &buffer)
+	if err != nil {
+		return nil, err
+	}
+	req.Header.Set("Content-Type", w.FormDataContentType())
+
+	return req, nil
+}
+
+func SendClaimFileToCollector(claimFilePath, createdBy, partnerName string) error {
```

Comment:
why not add the endpoint to this function's arguments and get the endpoint as an optional env variable

Reply:
Do you mean setting endpoint as an env variable back in suite_test?